File: Programazioa/JavaProiektua/Pruebas/pGarageDIY/src/myClasses/Worker.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package myClasses;

import java.time.LocalTime;

/**
 *
 * @author AdriAlex
 */
public class Worker {
    
    private static int id_Work = 1;
    
    private int id_worker;
    private String name;
    private String surname;
    private String password;
    private String occupation;
    private String mail;
    private int phone_Number;
    private double salary;
    private LocalTime start_Time;
    private LocalTime finish_Time;
    
    
    public Worker() {
        
    }
    
    public Worker (/*int id_worker,*/ String name, String surname, String password,
            String occupation, String mail, int phone_Number, double salary, 
            LocalTime start_Time, LocalTime finish_Time) {
        
        //this.id_worker = id_worker;
        this.id_worker = id_Work++;
        this.name = name;
        this.surname = surname;
        this.password = password;
        this.occupation = occupation;
        this.mail = mail;
        this.phone_Number = phone_Number;
        this.salary = salary;
        this.start_Time = start_Time;
        this.finish_Time = finish_Time;
        
    }
    
    public Worker (int id_worker, String name, String surname, String password,
            String occupation, String mail, int phone_Number, double salary, 
            LocalTime start_Time, LocalTime finish_Time) {
        
        this.id_worker = id_worker;
        this.name = name;
        this.surname = surname;
        this.password = password;
        this.occupation = occupation;
        this.mail = mail;
        this.phone_Number = phone_Number;
        this.salary = salary;
        this.start_Time = start_Time;
        this.finish_Time = finish_Time;
        
    }
    
    
    
    public int getId_worker() {
        return id_worker;
    }
    
    public String getName() {
        return name;
    }
    
    public void setName(String name) {
        this.name = name;
    }
    
    public String getSurname() {
        return surname;
    }
    
    public void setSurname(String surname) {
        this.surname = surname;
    }
    
    public String getPassword() {
        return password;
    }
    
    public void setPassword(String password) {
        this.password = password;
    }
    
    public String getOccupation() {
        return occupation;
    }
    
    public void setOccupation(String occupation) {
        this.occupation = occupation;
    }
    
    public String getMail() {
        return mail;
    }
    
    public void setMail(String mail) {
        this.mail = mail;
    }
    
    public int getPhone_Number() {
        return phone_Number;
    }
    
    public void setPhone_Number(int phone_Number) {
        this.phone_Number = phone_Number;
    }
    
    public double getSalary() {
        return salary;
    }
    
    public void setSalary(double salary) {
        this.salary = salary;
    }
    
    public LocalTime getStart_Time() {
        return start_Time;
    }
    
    public void setStart_Time(LocalTime start_Time) {
        this.start_Time = start_Time;
    }
    
    public LocalTime getFinish_Time() {
        return finish_Time;
    }
    
    public void setFinish_Time(LocalTime finish_Time) {
        this.finish_Time = finish_Time;
    }
    
    
    public String toString() {
        return "Worker: \n[id: " + this.getId_worker() + ", full name: " + this.getName() + " " + this.getSurname() + 
                ", password: " + this.getPassword() + ", occupation: " + this.getOccupation() + ", mail: " + this.getMail() + 
                ", phone number: " + this.getPhone_Number() + ", salary: " + this.getSalary() + 
                ", start time: " + this.getStart_Time().toString() + ", finish time: " + this.getFinish_Time() + "]";
    }
    
    
    
    public String toStringExtended() {
        return /*"\nWorker ID: " + this.getId_worker() + */
                "\nName: " + this.getName() + "\tSurname: " + this.getSurname() + 
                "\nOccupation: " + this.getOccupation() + "\tSalary: " + this.getSalary() + 
                "\nMail: " + this.getMail() + "\tPhone: " + this.getPhone_Number() + 
                "\nStart Time: " + this.getStart_Time() + "\tFinish Time: " + this.getFinish_Time() + 
                "\n--------------------------------------------------x--------------------------------------------------";
       
    }
    
    
    
}

File: GarapenIngurumenak/Python_Proyect/diyGaragePythonOOP/Worker.py
import time

# import Func_Worker
from Person import Person
import BasicMethodsToWorkWith
from datetime import datetime, time

allOccupations = ["Mechanic", "Painter", "Inspector"]


class Worker:
    def __init__(self):
        Person.__init__(self)

        self.worker_ID = BasicMethodsToWorkWith.BasicsMethods.askinteger("your Worker's ID")

        # validate the user entered value/data for the Worker's OCCUPATION
        validOccupation = 1
        while validOccupation == 1:
            occupationUser = BasicMethodsToWorkWith.BasicsMethods.askstring(
                "your OCCUPATION: " + "\n\t\t" + str(allOccupations)).capitalize()
            if occupationUser not in allOccupations:
                print("No result found for the specified occupation. Try again! \n")
            else:
                self.occupation = occupationUser
                validOccupation = 0

        # validate the user entered value/data for the Worker's SALARY
        validSalary = 1
        while validSalary == 1:
            salaryUser = BasicMethodsToWorkWith.BasicsMethods.askfloat("worker's SALARY")
            if salaryUser > 1500:
                print("The first salary cant be bigger than 1500.00 €. ")
            else:
                self.salary = salaryUser
                validSalary = 0

        # specify the STARTING and FINISHING time of working
        self.start_time = BasicMethodsToWorkWith.BasicsMethods.asktime("the time you START working").time()
        self.finish_time = BasicMethodsToWorkWith.BasicsMethods.asktime("the time you FINISH working").time()

    # getter methods
    def getWorkerID(self):
        return self.worker_ID

    def getName(self):
        return self.name

    def getSurname(self):
        return self.surname

    def getPassword(self):
        return self.password

    def getOccupation(self):
        return self.occupation

    def getMail(self):
        return self.mail

    def getPhoneNum(self):
        return self.phone_number

    def getSalary(self):
        return self.salary

    def getStartHour(self):
        return self.start_time

    def getFinishHour(self):
        return self.finish_time

    """
    It doesnt make much sence to change the main user data (like: Username, Name, Surname, Birthday); so 
    we will only give the user the option to modify/change those that may vary over time.

        -> Fixed data: Worker ID, Name, Surname, Mail
        -> Variable data: Salary, Occupation, Start Time, Finish Time, Password, Phone Number, 
    """

    # setter methods
    def setSalary(self):
        self.salary = BasicMethodsToWorkWith.BasicsMethods.askfloat("your new Salary: ")

    def setOccupation(self):
        self.occupation = BasicMethodsToWorkWith.BasicsMethods.askstring("your new OCCUPATION: ")

    def setStartTime(self):
        self.start_time = BasicMethodsToWorkWith.BasicsMethods.asktime("your new working START TIME ('hh:mm'): ")

    def setFinishTime(self):
        self.finish_time = BasicMethodsToWorkWith.BasicsMethods.asktime("your new working FINISH TIME ('hh:mm'): ")

    def printWorker(self):
        print("\t-> " + str(self.worker_ID) + ", " +
              str(self.name) + ", " +
              str(self.surname) + ", " +
              str(self.password) + ", " +
              str(self.occupation) + ", " +
              str(self.mail) + ", " +
              str(self.phone_Number) + ", " +
              str(self.salary) + ", " +
              str(self.start_time) + ", " +
              str(self.finish_time)
              )

    def printWorkerExtended(self):
        return str("\nWorker ID: ") + str(self.worker_ID) + ", " + str("Name: ") + str(self.name) + ", " + str(
            "Surname: ") + str(self.surname) + ", " + str("Password: ") + str(self.password) + ", " + str(
            "\nOccupation: ") + str(self.occupation) + ", " + str("Mail: ") + str(self.mail) + ", " + str(
            "Phone Number: ") + str(self.phone_Number) + str("\nSalary: ") + str(self.salary) + ", " + str(
            "Start Working: ") + str(self.start_time) + ", " + str("Finish Working: ") + str(self.finish_time) + str(
            "\n--------------------------------------------- \n ")
